main.go

```diff
@@ -3,19 +3,34 @@ package main
 import (
 	"context"
 	"fmt"
+	"time"
+	"sync"
 
 	"github.com/cloudchacho/hedwig-go"
 	"github.com/cloudchacho/hedwig-go/gcp"
 	"github.com/cloudchacho/hedwig-go/protobuf"
 	"google.golang.org/protobuf/proto"
 )
 
+
+const defaultVisibilityTimeoutS = time.Second * 20
+
+
 type ProcessSettings struct {
 	// interval when leader moves files to final bucket
 	ScrapeInterval int
 
 	// interval when follower flushes to staging bucket
 	FlushAfter int
+
+	// bucket where leader file is saved
+	MetadataBucket string
```

Comment:
moved these here since they are used in firehose top lvl

Reply:
I think it's better to move all the settings out of gcp and let it just be more of a gcp specific utility struct. I realized I need all those in leader implementation too.

Reply:
ah, I see you already have.